File: app/src/main/java/com/example/yelpbusinessexample/data/repository/BusinessRepository.java
package com.example.yelpbusinessexample.data.repository;

import com.example.yelpbusinessexample.data.model.Businesses;

import java.util.HashMap;
import java.util.List;

public interface BusinessRepository {

   interface LoadBusinessesCallback {
      void onBusinessesLoaded(List<Businesses> businesses,int total);
      void onDataNotAvailable();
      void onError(boolean isCancel);
   }

   void cancelCall();
   void getBusinesses(HashMap<String, String> queryMap);
}

File: app/src/main/java/com/example/yelpbusinessexample/data/repository/BusinessRepository.kt
package com.example.yelpbusinessexample.data.repository

import com.example.yelpbusinessexample.data.model.Businesses
import java.util.HashMap

interface BusinessRepository {

    interface LoadBusinessesCallback {
        fun onBusinessesLoaded(businesses: List<Businesses>, total: Int)
        fun onError(isCancel: Boolean)
    }

    fun cancelCall()
    fun getBusinesses(queryMap: HashMap<String, String>, callback: LoadBusinessesCallback)
}